date.h:6523:25 error: 'tH' may be used uninitialized

cmake fails to complete with
warzone2100/3rdparty/date/include/date/date.h:6523:25: error: 'tH' may be used uninitialized in this function [-Werror=maybe-uninitialized]
                         checked_set(H, tH, not_a_hour, is);
cc1plus: all warnings being treated as errors
src/CMakeFiles/warzone2100.dir/build.make:1386: recipe for target 'src/CMakeFile s/warzone2100.dir/updatemanager.cpp.o' failed
make[2]: *** [src/CMakeFiles/warzone2100.dir/updatemanager.cpp.o] Error 1
CMakeFiles/Makefile2:1591: recipe for target 'src/CMakeFiles/warzone2100.dir/all ' failed

**To Reproduce**
Steps to reproduce the behavior:
download source for 3.4.0 from sourceforge
extract tar.xz and cd into warzone dir
mkdir build
cd build
cmake   (with various options)
make
make install DESTDIR=/tmp/packagedir

**Expected behavior**
the project to complete compilation.

**Your System:**
 - OS: Linux (slackware 14.2)
 - Game version: [e.g. 3.4.0, sourceforge posted src xz]

**Additional context**
it looks like either the included date.h src header needs fixing or the cmake options should allow for warnings instead of halting at them (if that included header file is ok)

--- a/src/updatemanager.cpp
+++ b/src/updatemanager.cpp
@@ -37,8 +37,16 @@
 
 #include <sodium.h>
 #include <re2/re2.h>
+
+#if defined(__GNUC__) && !defined(__INTEL_COMPILER) && !defined(__clang__) && ((4 < __GNUC__) || ((4 == __GNUC__) && (7 <= __GNUC_MINOR__)))
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wmaybe-uninitialized" // Ignore on GCC 4.7+
+#endif
 #define ONLY_C_LOCALE 1
 #include <date/date.h>
+#if defined(__GNUC__) && !defined(__INTEL_COMPILER) && !defined(__clang__) && ((4 < __GNUC__) || ((4 == __GNUC__) && (7 <= __GNUC_MINOR__)))
+# pragma GCC diagnostic pop
+#endif
 
 #include <3rdparty/json/json.hpp>
 using json = nlohmann::json;

